python/cf_logging_enabled/cf_logging_enabled.py

```diff
@@ -0,0 +1,396 @@
+           As: a Security Officer
+       I want: CloudFront Distribution has been configured to store logs on an Authorized S3 bucket
+Scenarios:
+  Scenario 1:
+    Given: the CloudFront Distribution has not been configured to store logs on an Authorized S3 bucket
+     then: Return NON_COMPLIANT
+  Scenario 2:
+    Given: CloudFront Distribution has been configured to store logs
+      And: The S3 bucket where logs are stored is not the correctly authorized bucket
+     then: Return NON_COMPLIANT
+  Scenario 3:
+    Given: CloudFront Distribution has been configured to store logs on an authorized S3 bucket
+      And: The S3 bucket where logs are stored is the correctly authorized bucket
+     then: Return COMPLIANT
+'''
+
+import json
+import datetime
+import boto3
+import botocore
+
+##############
+# Parameters #
+##############
+
+# Define the default resource to report to Config Rules
+DEFAULT_RESOURCE_TYPE = ["AWS::CloudFront::Distribution"]
+
+# Set to True to get the lambda to assume the Role attached on the Config Service (useful for cross-account).
+ASSUME_ROLE_MODE = False
+
+#############
+# Main Code #
+#############
+
+def evaluate_compliance(event, configuration_item, valid_rule_parameters):
+    """Form the evaluation(s) to be return to Config Rules
+
+    Return either:
+    None -- when no result needs to be displayed
+    a string -- either COMPLIANT, NON_COMPLIANT or NOT_APPLICABLE
+    a dictionary -- the evaluation dictionary, usually built by build_evaluation_from_config_item()
+    a list of dictionary -- a list of evaluation dictionary , usually built by build_evaluation()
+
+    Keyword arguments:
+    event -- the event variable given in the lambda handler
+    configuration_item -- the configurationItem dictionary in the invokingEvent
+    valid_rule_parameters -- the output of the evaluate_parameters() representing validated parameters of the Config Rule
+
+    Advanced Notes:
+    1 -- if a resource is deleted and generate a configuration change with ResourceDeleted status, the Boilerplate code will put a NOT_APPLICABLE on this resource automatically.
+    2 -- if a None or a list of dictionary is returned, the old evaluation(s) which are not returned in the new evaluation list are returned as NOT_APPLICABLE by the Boilerplate code
+    3 -- if None or an empty string, list or dict is returned, the Boilerplate code will put a "shadow" evaluation to feedback that the evaluation took place properly
+    """
+
+    ###############################
+    # Add your custom logic here. #
+    ###############################
+    
+    logging_bucket = valid_rule_parameters ['CentralLoggingBucket'] + '.s3.amazonaws.com'
+    if not configuration_item['configuration']['distributionConfig']['logging']['enabled']:
+        return build_evaluation_from_config_item(configuration_item, 'NON_COMPLIANT', annotation='distribution is not configured to store logs')
+    else:

```

Comment:
This `else:` is not necessary, since the `if` statement is a `return`

Reply:
Updated the code

Reply:
updated the code